app/wrappers/wakes/facebook_metrics_wrapper.rb

```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+require 'json'
+require 'httparty'
+
+class Wakes::FacebookMetricsWrapper
+  include HTTParty
+  class FacebookError < StandardError; end
+  class FacebookNullResponse < StandardError; end
+  class FacebookRateLimitExceeded < StandardError; end
+  API_URL = 'https://graph.facebook.com/v2.5'
+  attr_reader :urls
+
+  def initialize(urls)
+    @urls = case urls
+            when String
+              [urls]
+            else
+              urls
+            end
+  end
+
+  def share_counts
+    {}.tap do |hash|
+      individual_responses.map do |r|
+        begin
+          hash[r['id']] = r.fetch('share').fetch('share_count')
+        rescue KeyError
+          # Some urls can have no 'share' keys. This is generally in the case when the url isn't valid.
+          Rails.logger.warn "Share count not received for #{r['id']}. Possibly an invalid url?"
+          hash[r['id']] = nil
+        end
+      end
+    end
+  end
+
+  private
+
+  def individual_responses
+    [].tap do |array|
+      parsed_batch_response.each do |individual_response|
+        raise FacebookNullResponse, 'one of the response is null' if individual_response.nil?
+        parsed_individual_response = JSON.parse(individual_response['body'])
+        array << parsed_individual_response
+      end
+    end
+  end
+
+  def parsed_batch_response
+    @parsed_response ||= begin
```

Comment:
@benhutton that might be left over from how the code used to be earlier. I can remove it.